zaza/model.py

```diff
@@ -1420,8 +1425,23 @@ def f(x):
     async def _unit_status():
         app = unit_name.split("/")[0]
         model_status = await async_get_status()
-        v = model_status.applications[app]['units'][unit_name][
-            'workload-status']['status']
+        if model_status.applications[app]['units']:
+            v = model_status.applications[app]['units'][unit_name][
+                'workload-status']['status']
+        else:
+            # For when the unit is a subordinate we need to get it's
+            # leader, and then get the status for the subordinate from it's
+            # unit
+            lead_app_name = subordinate_principal
+            if not subordinate_principal:
+                lead_app_name = model_status.applications[app][
+                    'subordinate-to'][0]
+            units = model_status.applications[lead_app_name]['units']
+            for unit in units:
+                if units[unit]['subordinates'].get(unit_name):
+                    v = units[unit]['subordinates'][unit_name][
+                        'workload-status']['status']
+                    break
```

Comment:
So this bit could be written a little bit more concisely as:

```python
for unit in units.values():
    try:
        v = unit['subordinates']['unit_name']['workload-status']['status']
        break
    except KeyError:
       pass
else:
    # Now we have an error situation as nothing was found for v?
    # raise an Exception?
```

As the comment in the code alludes to, there is also another issue with the code in that it doesn't handle `unit_name` not being found at all?  It's not clear from the function's docstring whether this is acceptable (i.e. returning None) or whether it should be an error?

Reply:
so, I updated this, suggested, but the KeyError is going to be an edge case, and would be difficult to cover in unit_tests. Unless you know of a scenario, I am more than happy to incorporate

Reply:
You could always use `# pragma: no cover` on those lines if the test is very awkward to write; but that would essentially mean that it's unreachable code?

Reply:
ok, I've added extra stuff to go through the first, as that would be reasonable. The second one would be really awkward and potentially unreachable